src/components/config_parser.cpp

```diff
@@ -11,12 +11,12 @@
 POLYBAR_NS
 
 config_parser::config_parser(const logger& logger, string&& file)
-    : m_log(logger), m_config(file_util::expand(file)) {}
+    : m_log(logger), m_config_file(file_util::expand(file)) {}
 
 config::make_type config_parser::parse(string barname) {
```

Comment:
You can't return a reference here because it will point to a local variable and is invalid as soon as the function returns.

We should probably just replace `make_type` with `config`

Reply:
fixed and removed the now unused `config::make_type` alias